capellambse/extensions/validation/_validate.py

```diff
@@ -0,0 +1,263 @@
+# SPDX-FileCopyrightText: Copyright DB Netz AG and the capellambse contributors
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+__all__ = [
+    "Category",
+    "ElementValidation",
+    "ModelValidation",
+    "register_rule",
+    "Result",
+    "Results",
+    "Rule",
+    "store_result",
+    "Type",
+    "Validation",
+    "Validator",
+]
+
+import abc
+import collections.abc as cabc
+import dataclasses
+import enum
+import typing as t
+
+from lxml import etree
+
+import capellambse
+from capellambse import helpers
+from capellambse.model import common as c
+
+
+class Category(enum.Flag):
+    """A category for a rule."""
+
+    REQUIRED = enum.auto()
+    RECOMMENDED = enum.auto()
+    SUGGESTED = enum.auto()
```

Comment:
For that use case, I'd rather have the search function accept more than one Category instance to search for, something like:

```python
def by_category(self, *categories: Category):
# or
def by_category(self, category: Category, *other_categories: Category):
```

(The latter would have the interpreter complain if no category was passed at all, while for the former we would have to add an explicit check. However, personally I actually prefer the former in this case, because it looks much cleaner and such a check isn't much overhead to maintain.)

`Flag` suggests that some uses of these instances are legal when they shouldn't be. And importantly, it also adds a "null" member (`Category(0)`) to the enumeration, which – as far as I can tell – doesn't make sense to use in any (currently considered) context.